src/Generators/TestsGenerator.php

```diff
@@ -53,7 +54,11 @@ protected function getInserts()
     {
         $arrayModels = [$this->model];
 
-        if ($this->classExists('models', 'User')) {
+        if (
+            $this->classExists('models', 'User')
+            && ($this->isFactoryExists('User')
+            || $this->isLegacyFactoryExists('User'))
+        ) {
```

Comment:
let's implement both legacy and new checks inside the `isFactoryExists` method

```suggestion
        if ($this->classExists('models', 'User') && $this->isFactoryExists('User')) {
```